pkg/capabilities/consensus/ocr3/reduce/reduce_aggregator.go

```diff
@@ -0,0 +1,392 @@
+package reduce
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/shopspring/decimal"
+	"google.golang.org/protobuf/proto"
+
+	ocrcommon "github.com/smartcontractkit/libocr/commontypes"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+)
+
+type AggregatorConfig struct {
+	// Static data that is added into every report
+	AdditionalData map[string]any `mapstructure:"additionalData"`
```

Comment:
Is this going to be eg. the FeedID ?

Reply:
I think if other fields are provided we should still try and aggregate them.

I see the config being slighltly different:
- a user provides a list of fields to aggregate along with its aggregation method.
  eg. FeedID - Mode, Price - Median, Timestamp - Median
- they also provide a field that we want to check the deviation logic against; in our case Price

Reply:
Yes exactly, you can see the CCIP Data Feeds use case recreated in [the uncommented test](https://github.com/smartcontractkit/chainlink-common/pull/842/files#diff-e3c476aac2b96c73893bd0d49244f6bc32b63e97c98c5876e6368000712209b8R210).

Yeahhh, that's fair - just run it all through aggregation. I had thought about that too. I'll make that change.